<a href="https://colab.research.google.com/github/jacobmlloyd/credit-risk-modeling/blob/main/Credit_Risk_Modeling_Potential_Sets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First Data Set (Approved)

This data set uses the Lending Club's publicly available data. The Bad column is our target, and it indicates if the loan ended up in bad standing. Below you can find the data dictionary. This will contain a classification model. Data taken from https://github.com/deepanshu88/Datasets/blob/master/CreditData/Loan%20Data.csv

| Name   | Description                                    |
|--------|------------------------------------------------|
| YOB    | Year of Birth (99 if unknown)                  |
| NKID   | Number of Children                             |
| DEP    | Number of Other Dependents                     |
| PHON   | Is there a homephone? 1 = yes; 0 = no          |
| SINC   | Spouses Income                                 |
| AES    | Applicants Employment Status (See Notes Below) |
| DAINC  | Applicants Income                              |
| RES    | Residential Status (See Notes Below)           |
| DHVAL  | Value of Home                                  |
| DMORT  | Outstanding Mortgage Balance                   |
| DOUTM  | Outgoings on Mortgage or Rent                  |
| DOUTL  | Outgoings on Loans                             |
| DOUTHP | Outgoings on Hire Purchase                     |
| DOUTCC | Outgoings on Credit Cards                      |
| BAD    | 1 = Bad, 0 = Good                              |

> AES Options: V = Government,
  W = housewife,
  M = military,
  P = private sector,
  B = public sector,
  R = retired,
  E = self employed,
  T = student,
  U = unemployed,
  N = others,
  Z  = no response

> RES Options: O = Owner,
  F = tenant furnished,
  U = Tenant Unfurnished,
  P = With parents,
  N = Other,
  Z = No response,

> DHVAL Options: 0 = no response or not owner,
  000001 = zero value,
  blank = no response,

> DMORT Options: 0 = no response or not owner,
  000001 = zero balance,
  blank = no response,

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("https://raw.githubusercontent.com/jacobmlloyd/credit-risk-modeling/main/Loan%20Data.csv")
df.head()

YOB  NKID  DEP  PHON  SINC AES    DAINC RES    DHVAL   DMORT  DOUTM  \
0  19.0   4.0  0.0     1   0.0   R      0.0   O  14464.0     4.0    0.0   
1  41.0   2.0  0.0     1   0.0   P  36000.0   O      0.0     0.0  280.0   
2  66.0   0.0  0.0     1   0.0   N  30000.0   N      0.0     0.0    0.0   
3  51.0   2.0  0.0     1   0.0   P    464.0   O  24928.0  8464.0  584.0   
4  65.0   0.0  0.0     1   0.0   P  15000.0   P      0.0     0.0    0.0   

   DOUTL  DOUTHP  DOUTCC  BAD  
0    0.0     0.0     0.0  0.0  
1  664.0     0.0    80.0  0.0  
2    0.0     0.0     0.0  0.0  
3  320.0     0.0    60.0  0.0  
4    0.0     0.0     0.0  0.0

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1225 entries, 0 to 1224
Data columns (total 15 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   YOB     1225 non-null   float64
 1   NKID    1225 non-null   float64
 2   DEP     1225 non-null   float64
 3   PHON    1225 non-null   int64  
 4   SINC    1225 non-null   float64
 5   AES     1225 non-null   object 
 6   DAINC   1225 non-null   float64
 7   RES     1225 non-null   object 
 8   DHVAL   1225 non-null   float64
 9   DMORT   1225 non-null   float64
 10  DOUTM   1225 non-null   float64
 11  DOUTL   1225 non-null   float64
 12  DOUTHP  1225 non-null   float64
 13  DOUTCC  1225 non-null   float64
 14  BAD     1225 non-null   float64
dtypes: float64(12), int64(1), object(2)
memory usage: 143.7+ KB


There are 1225 rows, and 14 features. Overall with this data set I see the biggest challenge being the fact that the data has so many values that are not normal, for example there are zero values represented as non-zero numbers.

## Second Data Set

This data set contains information about vehicles and their evaluation. More information on the dataset can be found here: http://archive.ics.uci.edu/ml/datasets/Car+Evaluation

In [12]:
df2 = pd.read_csv("https://raw.githubusercontent.com/jacobmlloyd/credit-risk-modeling/main/car.data", index_col=None, header=None)
df2.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1225 entries, 0 to 1224
Data columns (total 15 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   YOB     1225 non-null   float64
 1   NKID    1225 non-null   float64
 2   DEP     1225 non-null   float64
 3   PHON    1225 non-null   int64  
 4   SINC    1225 non-null   float64
 5   AES     1225 non-null   object 
 6   DAINC   1225 non-null   float64
 7   RES     1225 non-null   object 
 8   DHVAL   1225 non-null   float64
 9   DMORT   1225 non-null   float64
 10  DOUTM   1225 non-null   float64
 11  DOUTL   1225 non-null   float64
 12  DOUTHP  1225 non-null   float64
 13  DOUTCC  1225 non-null   float64
 14  BAD     1225 non-null   float64
dtypes: float64(12), int64(1), object(2)
memory usage: 143.7+ KB


Our target for this dataset is column 6, the actual evaluation. This is a multi-class classification problem. There are 5 features in this dataset and 1225 rows. I forsee predicting the evaluation off of such little data being difficult. I will have to hypertune to get anywhere near the accuracy I want. 